javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -61,6 +62,38 @@ object Integer {
   final val SIZE = 32
   final val BYTES = 4
 
+  private final val SignBit = Int.MinValue
+
+  /** Quantities in this class are interpreted as unsigned values.
+   *
+   *  - `maxLength`: `toUnsignedString(UInt.MaxValue, radix).length()`.
+   *    This is the maximum length of a string that can be parsed into an unsigned
+   *    `Int` using the given radix, assuming there is no leading sign or 0 digit.
+   *  - `overflowBarrier`: `divideUnsigned(UInt.MaxValue, radix)`.
+   *    This is the largest unsigned `x: Int` such that `x * radix` does not overflow.
+   */
+  private final class StringRadixInfo(val maxLength: Int, val overflowBarrier: Int)
+
+  /** Precomputed table for parseIntInternal. */
+  private lazy val StringRadixInfos: Array[StringRadixInfo] = {
```

Comment:
Did you compare this in code size and speed with two `Array[Int]`s? Especially since lookup in `overflowBarriers` would be rare, this might be worth it.

Reply:
Code size is unfortunately not meaningfully different. The extra `lazy val` and the extra loop to compute it compensate for the removed `class StringRadixInfo` (which is actually very small). IMO the code is less clean if we split into two arrays (see [diff](https://gist.github.com/sjrd/3d9d4ac0b55f34b2089aad68ec553c35)), so I lean towards keeping the `class StringRadixInfo`.